containers/podman-desktop
# chore: button component

### What does this PR do?

We currently use PatternFly classes like pf-c-button, pf-m-primary, pf-c-button__icon, and pf-m-start to style our buttons. Although they're powerful and we have a similar design, they tie us to PatternFly and its choices, and weren't really meant for desktop apps or dark mode. There are a few minor differences from our original design:
- slightly more button rounding
- in dark mode the hover should be lighter, not darker
- disabled buttons should be lighter
- spinner should replace button icon when there's progress
and this is repeated styles that aren't in a component.

This change adds a new Button component with support for styling, icons, and an inProgress spinner. It adds tests, and switches the buttons for the main lists, most form pages, and the MessageBox.

This is just a start to use the buttons in some core areas; after this PR there would need to be one or two more to replace all other instances.

### Screenshot/screencast of this PR

<img width="513" alt="Screenshot 2023-07-25 at 3 25 00 PM" src="https://github.com/containers/podman-desktop/assets/19958075/8a92ac23-19ce-4e61-a84e-a5e07910b5f7">

### What issues does this PR fix or reference?

N/A

### How to test this PR?

Go to all of these places and confirm that the buttons behave correctly, including disabled and in-progress states where applicable.
